javac-plugin/src/test/java/org/moditect/deptective/plugintest/basic/BasicPluginTest.java

```diff
@@ -81,6 +81,72 @@ public void shouldDetectInvalidClassAssignment() {
                 packageFooMustNotAccess("org.moditect.deptective.plugintest.basic.barclass"));
     }
 
+    @Test
+    public void shouldDetectInvalidQualifiedClassReference() {
+        Compilation compilation = compile();
+        assertThat(compilation).failed();
+
+        assertThat(compilation).hadErrorContaining(
```

Comment:
There should be two errors, one at the import and one at the reference. I suppose the test fails if you expect two errors.